File: webapp/src/main/java/io/ehdev/conrad/app/service/version/model/VersionCreateModel.java
package io.ehdev.conrad.app.service.version.model;

import io.ehdev.conrad.backend.commit.CommitDetails;
import io.ehdev.conrad.backend.commit.internal.DefaultCommitDetails;

import javax.validation.constraints.NotNull;
import java.util.List;

public class VersionCreateModel extends VersionSearchModel {
    @NotNull
    String message;

    @NotNull
    String commitId;

    public VersionCreateModel() {
    }

    public VersionCreateModel(List<String> commits, String message, String commitId) {
        super(commits);
        this.message = message;
        this.commitId = commitId;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getCommitId() {
        return commitId;
    }

    public void setCommitId(String commitId) {
        this.commitId = commitId;
    }

    public CommitDetails toCommitDetails() {
        return new DefaultCommitDetails(commitId, message);
    }
}

File: api-model/src/main/kotlin/io/ehdev/conrad/model/version/VersionCreateModel.kt
package io.ehdev.conrad.model.version

import com.fasterxml.jackson.annotation.JsonCreator
import javax.validation.constraints.NotNull

class VersionCreateModel public @JsonCreator constructor(
    commits: List<String>,
    @NotNull val message: String,
    @NotNull val commitId: String,
    @NotNull val token: String
): VersionSearchModel(commits);
